RileyLinkKitTests/PumpOpsSynchronousTests.swift

```diff
@@ -73,202 +82,351 @@ class PumpOpsSynchronousTests: XCTestCase {
         
         let events:[PumpEvent] = [squareWaveBolus]
         
-        let (timeStampedEvents, _) = sut.convertPumpEventToTimestampedEvents(pumpEvents: events, startDate: Date.distantPast, pumpModel: pumpModel)
+        let (timeStampedEvents, _, _) = sut.convertPumpEventToTimestampedEvents(pumpEvents: events, startDate: Date.distantPast, pumpModel: pumpModel)
         
         // It should be included
         XCTAssertTrue(array(timeStampedEvents, containsPumpEvent: squareWaveBolus))
     }
+    
+    func testEventAfterDelayedAppendEventFor522IsReturned() {
+        setUpTestWithPumpModel(.Model522)
+        
+        let tempEventBasal = createTempEventBasal2016()
+        let events:[PumpEvent] = [createSquareBolusEvent2010(), tempEventBasal]
+        let (timeStampedEvents, _, _) = sut.convertPumpEventToTimestampedEvents(pumpEvents: events, startDate: Date.distantPast, pumpModel: pumpModel)
+        
+        // It should be returned (can't tell if the time for the SquareBolus is valid
+        assertArray(timeStampedEvents, containsPumpEvent: tempEventBasal)
+    }
+    
+    func testEventAfterDelayedAppendEventFor523IsNotReturned() {
+        setUpTestWithPumpModel(.Model523)
+        
+        let tempEventBasal = createTempEventBasal2016()
+        let events:[PumpEvent] = [createSquareBolusEvent2010(), tempEventBasal]
+        let (timeStampedEvents, _, _) = sut.convertPumpEventToTimestampedEvents(pumpEvents: events, startDate: Date.distantPast, pumpModel: pumpModel)
+        
+        // It should not be returned (timestamp from square bolus is valid)
+        assertArray(timeStampedEvents, doesntContainPumpEvent: tempEventBasal)
+    }
 
+    func testDelayedAppendDiscontinuityEventFor522IsReturned() {
+        setUpTestWithPumpModel(.Model522)
+        
+        let squareBolus2016 = createSquareBolusEvent2016()
+        let events:[PumpEvent] = [createSquareBolusEvent2010(), squareBolus2016]
+        let (timeStampedEvents, _, _) = sut.convertPumpEventToTimestampedEvents(pumpEvents: events, startDate: Date.distantPast, pumpModel: pumpModel)
+        
+        //It should be returned
+        XCTAssertTrue(array(timeStampedEvents, containsPumpEvent: squareBolus2016))
+    }
+    
+    func test523SquareBolusEventIsNotReturned() {
+        setUpTestWithPumpModel(.Model523)
+        
+        let squareBolus2016 = createSquareBolusEvent2016()
+        let events:[PumpEvent] = [createSquareBolusEvent2010(), createSquareBolusEvent2010(), squareBolus2016]
+        let (timeStampedEvents, _, _) = sut.convertPumpEventToTimestampedEvents(pumpEvents: events, startDate: Date.distantPast, pumpModel: pumpModel)
+        
+        //It should not be returned
+        XCTAssertFalse(array(timeStampedEvents, containsPumpEvent: squareBolus2016))
+    }
+    
+    func test523RegularBolusDoesReturnEvent() {
+        setUpTestWithPumpModel(.Model523)
```

Comment:
In this test, the events look like a pump date discontinuity, but since the event signaling the discontinuity is a delayed append event, we return it.  

This is the same condition and test set up in testDelayedAppendDiscontinuityEventFor522IsReturned, so it should be removed.